src/withResolution/mediaQuery.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+type Callback = (mql: MediaQueryList, alias: string) => void
+type QueryData = {
+  [query: string]: {|
```

Comment:
You could replace this with
```
ReadOnly<{
  mediaQuery: MediaQueryList,
  allCallbacks: (mql: MediaQueryListEvent) => void,
}>
```

Reply:
the values themselves are read only once set but the array itself is not read only